src/Atc.Cosmos.EventStore.Cqrs/Projections/ProjectionProcessor.cs

```diff
@@ -45,11 +43,13 @@ public async Task<ProjectionAction> ProcessBatchAsync(
 
         using var batchTelemetry = telemetry.StartBatch(projectionName, groupedEvents.Length);
 
-        var projection = projectionFactory
-            .GetProjection<TProjection>();
-
         foreach (var events in groupedEvents)
         {
+            var pf = projectionFactory.CreateScope();
+
+            var projection = pf.GetProjection<TProjection>();
+            var projectionMetadata = pf.GetProjectionMetadata<TProjection>();
```

Comment:
`ProjectionMetadata` is designed to be a singleton as its sole purpose is to build up an internal dictionary of `MethodInfo` so when an event needs to be consumes by a projection it's does so fast with no need for reflection. The reflection is done on the type of projection (`TProjection`) and required no instance data besides the lookup table, when calling the projection the actual instance of the projection is passed in.

This should be move back into the constructor.

Reply:
That makes a lot of sense. 
Is there any reason that the factory is creating it instead of it being injected?

Reply:
You are right, it should be possible to do when registering the projection.